package freerds.icp;

enum MSGTYPE {
	IsChannelAllowed = 1;
	Ping             = 2;
	GetUserSession   = 3;
	DisconnectUserSession = 4;
	LogOffUserSession = 5;
	FdsApiVirtualChannelOpen = 6;
	LogonUser = 7;
}

message IsChannelAllowedRequest {
	required string ChannelName = 1;	
}
message IsChannelAllowedResponse {
	required bool ChannelAllowed = 1;
}

message PingRequest {
}

message PingResponse {
	required bool pong = 1;
}

message GetUserSessionRequest {
	required string username = 1;
	optional string domainname = 2;
}

message GetUserSessionResponse {
	required uint32 SessionID = 1;
	required string ServiceEndpoint = 2;
}

message DisconnectUserSessionRequest {
	required uint32 SessionID = 1;
}

message DisconnectUserSessionResponse {
	required bool disconnected = 1;
}

message LogOffUserSessionRequest {
	required uint32 SessionID = 1;
}

message LogOffUserSessionResponse {
	required bool loggedoff = 1;
}

message FdsApiVirtualChannelOpenRequest {
	required uint32 SessionID = 1;
	required string VirtualName = 2;
}

message FdsApiVirtualChannelOpenResponse {
	required string ConnectionString = 1;
}

message LogonUserRequest {
	required uint32 SessionId = 1;
	required string Username = 2;
	required string Password = 3;
	required string Domain = 4;
}

message LogonUserResponse {
	required uint32 SessionId = 1;
	required int32 AuthStatus = 2;
	required string ServiceEndpoint = 3;
}
